app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 class PostsController < ApplicationController
   def index
     @user = User.includes(posts: [:comments]).find(params[:user_id])
-    @posts = Post.where(user_id: @user.id).find_by(params[:id])
+    @posts = Post.where(user_id: @user.id)
+  end
+
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      flash[:success] = 'Post Added Successfully'
+      redirect_to user_posts_path(current_user)
+    else
+      flash.now[:error] = 'Post could not be added'
+    end
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```

Comment:
**OPTIONAL** - It is advisable to make this method a private method, you can read this [article](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) for more information